docs/guide/writing-tasks.md

```diff
@@ -173,8 +173,9 @@ If your dependencies are updated often, please pay attention to `fingerprint_scr
 
 `reupload_on_changes` is an *optional* field that can specify whether Cirrus Agent should check if 
 contents of cached `folder` have changed during task execution and re-upload a cache entry in case of any changes.
-`reupload_on_changes` option is enabled by default and Cirrus Agent will detect additions, deletions and modifications
-of any files under specified `folder`. All of the detected changes will be logged under `Upload '$CACHE_NAME' cache`instructions for easier debugging of cache invalidations.
+If `reupload_on_changes` option is not set explicitly then it will be set to `false` if `fingerprint_script` is presented and `true` tohervise.
```

Comment:
```suggestion
If `reupload_on_changes` option is not set explicitly then it will be set to `false` if `fingerprint_script` is presented and `true` otherwise.
```